tests/poll.rs

```diff
@@ -310,6 +310,43 @@ fn registry_operations_are_thread_safe() {
     handle3.join().unwrap();
 }
 
+#[test]
+fn register_during_poll() {
+    let (mut poll, mut events) = init_with_poll();
+
+    let registry = Arc::new(poll.registry().try_clone().unwrap());
+
+    let barrier = Arc::new(Barrier::new(2));
+    let barrier1 = Arc::clone(&barrier);
+
+    let handle1 = thread::spawn(move || {
+        let mut stream = UdpSocket::bind(any_local_address()).unwrap();
+
+        barrier1.wait();
```

Comment:
Thoughts on adding some millisecond sleep here? That way both threads unblock at the barrier. The poll thread immediately goes into a 5 second poll, and this thread waits just a little bit so we can get closer to "trying" to register during a poll.